projects/keycloak-angular/src/lib/core/services/keycloak.service.ts

```diff
@@ -68,6 +69,18 @@ export class KeycloakService {
    */
   private _keycloakEvents$: Subject<KeycloakEvent> =
     new Subject<KeycloakEvent>();
+  /**
+   * The amount of required time remaining before expiry of the token before the token will be refreshed.
+   */
+  private _updateMinValidity: number;
+  /**
+   * Returns true if the request should have the token added to the headers by the KeycloakBearerInterceptor.
+   */
+  shouldAddToken: (_: HttpRequest<any>) => Boolean;
+  /**
+   * Returns true if the request being made should potentially update the token.
+   */
+  shouldUpdateToken: (_: HttpRequest<any>) => Boolean;
```

Comment:
We can also add much stricter types here for each of the functions by injecting the full definition so users get a better DX.

```suggestion
  /**
   * Returns true if the request should have the token added to the headers by the KeycloakBearerInterceptor.
   */
  shouldAddToken: (_: HttpRequest<any>) => Boolean;
  /**
   * Returns true if the request being made should potentially update the token.
   */
  shouldUpdateToken: (_: HttpRequest<any>) => Boolean;
```

Reply:
Can you elaborate what you mean by that?

Reply:
I mean that if we know the response body shape then we should create a custom type and set it instead of `any` so we get proper typing and IDE support, etc, does that make sense? If we cannot know the shape of the response body then `unknown` could be a route to be clear about that. I don't know. WDYT?